sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -354,9 +355,14 @@ resource
     : identifier STRING
     ;
 
+insertStatement
+    : (ctes)? insertInto queryTerm queryOrganization                                       #singleInsertQuery
+    | (ctes)? fromClause multiInsertQueryBody+                                             #multiInsertQuery
+    ;
+
 queryNoWith
-    : insertInto? queryTerm queryOrganization                                              #singleInsertQuery
-    | fromClause multiInsertQueryBody+                                                     #multiInsertQuery
+    : queryTerm queryOrganization                                               #noWithQuery
```

Comment:
Actually, `queryNoWith` has these two pattern. And both doesn't have `With`. Can we revise this name `#noWithQuery` more narrowly?

Reply:
@dongjoon-hyun 
```
queryNoWith
    : queryTerm queryOrganization                                               #noWithQuery
    | fromClause querySpecification queryOrganization                           #queryWithFrom
    ;
```
the #noWithQuery is taken up by the first alternative ? Any other preference ?

Reply:
@dongjoon-hyun would you have any suggestion ? I was thinking `selectQuery` but  this matches select, table query, inline table etc. So a little confused :-)